optexp/datasets/wikitext103.py

```diff
@@ -0,0 +1,172 @@
+import os
+import shutil
+from pathlib import Path
+
+import pyarrow.parquet
+import requests
+import torch
+from attrs import frozen
+from torch.utils.data import DataLoader
+
+from optexp.config import Config
+from optexp.datasets import Dataset
+from optexp.datasets.dataset import TrVa, HasClassCounts, Downloadable
+from optexp.datasets.tokenizers import Tokenizer, BPETokenizer
+
+
+#  TODO <unk> considered harmful
+@frozen
+class WikiText103(Dataset, HasClassCounts, Downloadable):
+
+    sequence_length: int = 1024
+    vocab_size: int = 50257
+    tokenizer: Tokenizer = BPETokenizer()
+    directory: Path = Config.get_dataset_directory() / "WikiText103"
```

Comment:
Why is the directory part of the experiment definition instead of a global variable for the file? This would lead to different experiment definition/hash if run on different machines

Reply:
changed